data/mods/ssb/moves.ts

```diff
@@ -1205,12 +1205,10 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
 		secondary: {
 			chance: 20,
 			onHit(target, source) {
-				switch(toID(source.types[0])) { 
-					case 'normal':
-					const typeList = ["Fighting", "Flying", "Poison", "Ground", "Rock",
-							"Bug", "Ghost", "Steel", "Fire", "Water", "Grass", "Electric",
-							"Psychic", "Ice", "Dragon", "Dark", "Fairy"];
-					const newType = typeList[this.random(typeList.length)];
+				switch (toID(source.types[0])) {
+				case 'normal':
+					const typeList = Object.keys(this.dex.data.TypeChart);
```

Comment:
I did test this, it didn't work and caused crashes afaik

Reply:
It crashed because I accidentally put Object.values and not Object.keys last time

Reply:
Lemme try if this will work

Reply:
Ohhh that makes sense